GitUI/CommandsDialogs/FormSettings.cs

```diff
@@ -112,6 +112,7 @@ public FormSettings([NotNull] GitUICommands commands, SettingsPageReference init
             settingsTreeView.AddSettingsPage(SettingsPageBase.Create<PluginRootIntroductionPage>(this), pluginsPageRef, icon: null, asRoot: true);
 
             var pluginEntries = PluginRegistry.Plugins
+                .Where(p => p.GetSettings().Any())
```

Comment:
Have you profiled this to see if calling `GetSettings` on all plugins is a quick operation?